pandas-stubs/core/generic.pyi

```diff
@@ -69,7 +69,7 @@ class NDFrame(PandasObject, indexing.IndexingMixin):
     def pop(self, item: _str) -> NDFrame: ...
     def squeeze(self, axis=...): ...
     def equals(self, other: Series[S1]) -> _bool: ...
-    def __neg__(self) -> None: ...
+    def __neg__(self: T) -> T: ...
```

Comment:
Is pos also wrong?

Reply:
I think pyright is picky about using too wide TypeVar's for self. Would be better to use `NDFrameT` instead of `T`

Reply:
Should be `def __neg__(self) -> NDFrame: ...`

Reply:
This means to behave like the `Self` typing in python 3.11 (it seems mypy doesn't support it).
So that you don't have to implement it for every subclass to return itself.
For me pyright worked fine (`T` will resolve to a subclass of `NDFrame` in any case because this definition is only for classes that derive from `NDFrame`)

Reply:
If you use the `def __neg__(self) -> NDFrame: ...` pattern, it will be consistent with the rest of the stubs

Reply:
But then I'll have to also add it to `Series` and `DataFrame`. If you think its better than I'll do it, but I think this way is less error prune (no "code duplication").

Reply:
Sorry, should be `def __neg__(self: NDFrameT) -> NDFrameT: ...` .  That should work and not require entries for `Series` and `DataFrame`